pkg/handler/v1/project/project.go

```diff
@@ -36,7 +37,7 @@ import (
 )
 
 // CreateEndpoint defines an HTTP endpoint that creates a new project in the system
-func CreateEndpoint(projectProvider provider.ProjectProvider) endpoint.Endpoint {
+func CreateEndpoint(projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, settingsProvider provider.SettingsProvider, memberMapper provider.ProjectMemberMapper, memberProvider provider.ProjectMemberProvider, userProvider provider.UserProvider) endpoint.Endpoint {
```

Comment:
This method is quite long right now, can you extract validation part to the new one?